lib/python/picongpu/input/parameters.py

```diff
@@ -147,6 +165,19 @@ def get_value(self):
         """
         return self.value
 
+    def as_dict(self):
+        """
+        Convert Parameter object into a plain dictionary
+
+        Returns
+        --------
+        A dictionary with the member variables as (key, value) pairs.
+        """
+
+        members = super(UiParameter, self).as_dict()
+        del members["formatter"]
```

Comment:
Seems like we need to exclude the "dtype" member as well since it can't be serialzed either

Reply:
It just occured to me that we could store the the dtype as a string representation using
`members["dtype"] = members["dtype"].__name__`. Maybe it would be cleaner to do this for later restoring the parameter objects from the json representation correctly.

Reply:
ok, but careful that a lambda does not mean it is just *one* source line ;)

Reply:
shall we add the string representation of both in a follow-up? do we need it now?
otherwise feel free to approve the PR :)

Reply:
I think for the moment we can leave them out, but maybe the class of each parameter object should be stored as a string representation.
This is because, as of right now, we still have issues in our ui-code when reading back the parameter variables from json: there, we implicitely have to cast to the base class "Parameter" (since we don't have any other information) losing all info concerning the dtype and formatter which the program then tries to access, of course crashing.
So I still need to figure out a proper way to restore an "exact" copy of the parameter object on our side of the code. But the dtype and the formatter are handled with defaults when not given, so that should be not a problem.

Reply:
Hold on, sorry...maybe our bug is somewhere else and we don't need the class representation. Don't change it just yet, I will get back to you :)

Reply:
I found the critical section: calling `del members["formatter"]` seems to remove the entry not only from the members dictionary, but since it is a reference also from the true object.